superset/queries/saved_queries/commands/importers/v1/__init__.py

```diff
@@ -0,0 +1,75 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Dict, Set
+
+from marshmallow import Schema
+from sqlalchemy.orm import Session
+
+from superset.queries.saved_queries.commands.exceptions import SavedQueryImportError
+from superset.commands.importers.v1 import ImportModelsCommand
+from superset.connectors.sqla.models import SqlaTable
+from superset.databases.commands.importers.v1.utils import import_database
+from superset.datasets.commands.importers.v1.utils import import_dataset
+from superset.datasets.schemas import ImportV1DatasetSchema
+from superset.queries.saved_queries.dao import SavedQueryDAO
+from superset.queries.saved_queries.commands.importers.v1.utils import import_saved_query
+from superset.queries.saved_queries.schemas import ImportV1SavedQuerySchema
+
+class ImportSavedQueriesCommand(ImportModelsCommand):
+    """Import Saved Queries"""
+
+    dao = SavedQueryDAO
+    model_name= "saved_queries"
+    prefix ="saved_queries/"
+    schemas: Dict[str, Schema] = {
+        "datasets/": ImportV1DatasetSchema(),
+        "queries/": ImportV1SavedQuerySchema(),
+    }
+    import_error = SavedQueryImportError
+
+    @staticmethod
+    def _import(
+        session: Session, configs: Dict[str, Any], overwrite: bool = False
+    ) -> None:
+        # discover databases associated with saved queries
+        database_uuids: Set[str] = set()
+        for file_name, config in configs.items():
+            if file_name.startswith("queries/"):
+                database_uuids.add(config["database_uuid"])
+
+        # import related databases
+        database_ids: Dict[str, int] = {}
+        for file_name, config in configs.items():
+            if file_name.startswith("databases/") and config["uuid"] in database_uuids:
+                database = import_database(session, config, overwrite=False)
+                database_ids[str(database.uuid)] = database.id
+
+
+        # import saved queries with the correct parent ref
+        for file_name, config in configs.items():
+            if file_name.startswith("queries/") and config["database_uuid"] in database:
+                # update datasource id, type, and name
+                database = database[config["dataset_uuid"]]
+                config.update(
+                    {
+                        "datasource_id": database.id,
+                        "datasource_name": database.table_name,
+                    }
+                )
+                config["params"].update({"datasource": database.uid})
```

Comment:
A saved query has a many-to-one relationship to database; there are no datasources involved. What you need to do here is update the [`db_id` attribute of the saved query model](https://github.com/apache/superset/blob/master/superset/models/sql_lab.py#L179) and point it to the imported/existing database:

```suggestion
                config["db_id"] = database_ids[config["database_uuid"]]
```

Reply:
That makes sense, thank you for showing me the source of what I am supposed to be looking at, I appreciate it. Quick clarifying question: `config["db_id"] = database_ids[config["database_uuid"]]` should come right after config["params"] (on line 74). right? I think this is a weird github thing because it shows the line after params, but the line number is 66.

Reply:
The change I suggested deletes lines 71-79, since datasources are not used in saved queries and replaces them with `config["db_id"] = database_ids[config["database_uuid"]]`.

You should be able to click "Commit suggestion" to automatically apply the suggested change, but then you need to pull the changes to your local branch if you need to modify it again. Or is the button not there for you?